pkg/api/v1/schema.go

```diff
@@ -637,6 +637,10 @@ type Interface struct {
 	// Logical name of the interface as well as a reference to the associated networks
 	// Must match the Name of a Network
 	Name string `json:"name"`
+	// Interface model supported by libvirt
+	// Defaults to virtio
+	// TODO:(ihar) should we enumerate and enforce allowed values
```

Comment:
can we do that in this patch series as part of the merge criteria?

Reply:
We may, I just am not sure if that's what we want to do. The API document doesn't list additional requirements, and this is already exposed by the annotations. @fabiand @vladikr do we want to limit possible values for model, and if so, what's the accepted list? (Obviously should include virtio, e1000, e1000e, rtl*, anything else?)

Reply:
@booxer we could. libvirt by default will let us only set:
`ne2k_isa i82551 i82557b i82559er ne2k_pci pcnet rtl8139 e1000 virtio` and in nova we explicitly support only: 
```
        'kvm': [network_model.VIF_MODEL_VIRTIO,                                 
                network_model.VIF_MODEL_NE2K_PCI,                               
                network_model.VIF_MODEL_PCNET,                                  
                network_model.VIF_MODEL_RTL8139,                                
                network_model.VIF_MODEL_E1000]
```
I guess we can start with these.

Reply:
The enumeration from openstack looks great as a start.

Reply:
In addition to that list we probably want e1000e - due to q35.